src/mscorlib/src/System/String.Searching.cs

```diff
@@ -113,15 +113,6 @@ public partial class String
         //
         [Pure]
         public int IndexOf(String value, int startIndex, int count) {
-            if (startIndex < 0 || startIndex > this.Length) {
-                throw new ArgumentOutOfRangeException(nameof(startIndex), Environment.GetResourceString("ArgumentOutOfRange_Index"));
-            }
-
-            if (count < 0 || count > this.Length - startIndex) {
-                throw new ArgumentOutOfRangeException(nameof(count), Environment.GetResourceString("ArgumentOutOfRange_Count"));
-            }
-            Contract.EndContractBlock();
```

Comment:
Note that removing the validation here does introduce a subtle behavior difference: if `value` is null, `ArgumentNullException` will be thrown now before the other arguments are validated. I'm guessing this is an acceptable change, but let me know if not and I'll revert this part of the change.